src/feed-text-sanitization.c

```diff
@@ -78,7 +81,10 @@ trim_to_first_n_sentences_under_threshold (const gchar *str,
       if (sentence_length == 0)
         continue;
 
-      if ((character_count + sentence_length_with_period) > character_threshold)
+      /* Even if the sentence would be longer than the character threshold
+       * we should always take it. We'll ellipsize it later at a word boundary. */
+      if ((character_count + sentence_length_with_period) > character_threshold &&
+          sentences->len > 0)
```

Comment:
Does this not go in the next commit, where you do the ellipsization?

Reply:
This should go in the first commit, since it fixes an issue with no sentences being taken at all if the first sentence was too long (though maybe I will remove the comment about ellipsization)